### PR TITLE
device plugin authoring guide

### PR DESCRIPTION
**Pull request by cgbaker**
added the device plugin authoring guide, made minor formatting changes to task driver plugin authoring guide.

### CODE REVIEW

File: website/source/docs/internals/plugins/devices.html.md [hunk 1]
```diff
@@ -3,10 +3,80 @@ layout: "docs"
...
+### Lifecycle and State
+
+A device plugin is long-lived. Nomad will ensure that one instance of the plugin is 
+running. If the plugin crashes or otherwise terminates, Nomad will launch another 
+instance of it. 
+
+However, unlike [task drivers](task-drivers.html), device plugins do not currently
+have an interface for persisting state to the Nomad client. Instead, the device
+plugin API emphasizes fingerprinting devices and reporting their status. After
+helping to provision a task with a scheduled device, they do not have any charter
```
Comment (lgfa29) [inline website/source/docs/internals/plugins/devices.html.md]: I think `charter` here is a bit too complex. Something use something like `access` or `permission`.

File: website/source/docs/internals/plugins/devices.html.md [hunk 2]
```diff
@@ -3,10 +3,80 @@ layout: "docs"
...
+Each fingerprint response consists of either an error or a list of device groups.
+A device group is a list of detected devices that are identical for the purpose of
+scheduling; that is, they will have identical attributes.
+
+### `Stats(context.Context, time.Duration) (<-chan *StatsResponse, error)`
+
+The `Stats` [function][statsFn] returns a channel on which the plugin should 
+emit device statistics, at the specified interval, until either an error is 
+encountered or the specified context is cancelled. The `StatsReponse` object
+allows dimensioned statistics to be returned for each device in a device group.
```
Comment (lgfa29) [inline website/source/docs/internals/plugins/devices.html.md]: What do you mean by `dimensioned statistics` here?

File: website/source/docs/internals/plugins/devices.html.md (hunk 1 above)
Comment (cgbaker) [reply website/source/docs/internals/plugins/devices.html.md]: i was thinking "charter" more in line with "prerogative" or "responsibility". how about the latter?

File: website/source/docs/internals/plugins/devices.html.md (hunk 2 above)
Comment (cgbaker) [reply website/source/docs/internals/plugins/devices.html.md]: "not unit-less"

Comment (cgbaker) [reply website/source/docs/internals/plugins/devices.html.md]: i'll add a clarifying link

File: website/source/docs/internals/plugins/devices.html.md (hunk 1 above)
Comment (lgfa29) [reply website/source/docs/internals/plugins/devices.html.md]: Yes, "responsibility" is good here

**Pull request merged**